src/granite/comparator.cr

```diff
@@ -0,0 +1,32 @@
+# Adds equals, hash, and comparison methods to Granite::Base objects.
+module Granite::Comparator
+  macro __process_comparator
+
+    def_equals_and_hash {{ *FIELDS.keys }}
```

Comment:
Both `Comparable` and `#def_equals_and_hash` define `#==`, so I think this line is redundant.